src/llscan.cc

```diff
@@ -1141,6 +1163,56 @@ uint64_t FindJSObjectsVisitor::Visit(uint64_t location, uint64_t word) {
 }
 
 
+void FindJSObjectsVisitor::InsertOnMapsToInstances(
+    uint64_t word, v8::Map map, FindJSObjectsVisitor::MapCacheEntry map_info,
+    v8::Error& err) {
+  TypeRecord* t;
+
+  /* No entry in the map, create a new one. */
+  if (llscan_->GetMapsToInstances().count(map_info.type_name) == 0) {
+    t = new TypeRecord(map_info.type_name);
+    llscan_->GetMapsToInstances().emplace(map_info.type_name, t);
+  } else {
+    /* Get entry from the map. */
+    t = llscan_->GetMapsToInstances().at(map_info.type_name);
+  }
+  /* Determine if this is a new instance.
+    * (We are scanning pointers to objects, we may have seen this location
+    * before.)
+    */
+  if (t->GetInstances().count(word) == 0) {
+    t->AddInstance(word, map.InstanceSize(err));
+  }
+}
+
+void FindJSObjectsVisitor::InsertOnDetailedMapsToInstances(
+    uint64_t word, v8::Map map, FindJSObjectsVisitor::MapCacheEntry map_info,
+    v8::Error& err) {
+  DetailedTypeRecord* t;
+
+  auto type_name_with_properties = map_info.GetTypeNameWithProperties();
+  /* No entry in the map, create a new one. */
+  if (llscan_->GetDetailedMapsToInstances().count(*type_name_with_properties) ==
+      0) {
+    t = new DetailedTypeRecord(*map_info.GetTypeNameWithProperties(false, 3),
+                               map_info.own_descriptors_count,
+                               map_info.indexed_properties_count);
+    llscan_->GetDetailedMapsToInstances().emplace(*type_name_with_properties,
+                                                  t);
+  } else {
+    /* Get entry from the map. */
```

Comment:
Feel free to drop these comments, it's almost at the level of `i += 1;   // add 1 to i`